cip-spec.md

```diff
@@ -0,0 +1,406 @@
+
+## Utxos
+
+### Asset
+
+Get all UTxOs that contain some of the specified asset
+
+#### Request
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"asset_name":"333333333333","minting_policy_hash":"33333333333333333333333333333333333333333333333333333333"}
+```
+</details>
+
+#### Response
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"utxos":[{"input":{"transaction_id":"3333333333333333333333333333333333333333333333333333333333333333","index":858993459},"output":{"address":"stake177stake177","amount":{"coin":"0000000000"},"script_ref":{"tag":"plutus_script","value":{"language":"plutus_v1/plutus_v1","bytes":"333333333333"}}}}]}
+```
+</details>
+
+### Transaction Hash
+
+Get all UTxOs produced by the transaction [These are only outputs, should they be in/outs??]
+
+#### Request
+
+<details>
+<summary>Show example: </summary>
+
+```
+"3333333333333333333333333333333333333333333333333333333333333333"
+```
+</details>
+
+#### Response
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"utxos":[{"input":{"transaction_id":"3333333333333333333333333333333333333333333333333333333333333333","index":858993459},"output":{"address":"stake177stake177","amount":{"coin":"0000000000"},"script_ref":{"tag":"plutus_script","value":{"language":"plutus_v1/plutus_v1","bytes":"333333333333"}}}}]}
+```
+</details>
+
+### Address
+
+Get all UTxOs present at the address
+
+#### Request
+
+<details>
+<summary>Show example: </summary>
+
+```
+"stake177stake177"
+```
+</details>
+
+#### Response
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"utxos":[{"input":{"transaction_id":"3333333333333333333333333333333333333333333333333333333333333333","index":858993459},"output":{"address":"stake177stake177","amount":{"coin":"0000000000"},"script_ref":{"tag":"plutus_script","value":{"language":"plutus_v1/plutus_v1","bytes":"333333333333"}}}}]}
+```
+</details>
+
+### Payment Credential
+
+Get all UTxOs present at the addresses which use the payment credential
+
+#### Request
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"tag":"pubkey_hash","value":"33333333333333333333333333333333333333333333333333333333"}
+```
+</details>
+
+#### Response
+
+<details>
+<summary>Show example: </summary>
+
+```
+{"utxos":[{"input":{"transaction_id":"3333333333333333333333333333333333333333333333333333333333333333","index":858993459},"output":{"address":"stake177stake177","amount":{"coin":"0000000000"},"script_ref":{"tag":"plutus_script","value":{"language":"plutus_v1/plutus_v1","bytes":"333333333333"}}}}]}
```

Comment:
Instead of example values (that are not valid anyway), show the schema or a link to the schema file.

Reply:
@klntsky I pushed an update inlining the full schema in the generated markdown. This unfortunately makes the file so big that even GH will not generate a markdown preview from it (locally, my editor takes almost a full minute to generate the preview :D).
I think inlining the schema (at least fully-resolved schemas) is a no-go. I like the idea of linking to the schema file we discussed and will try to explore some options with that (ideally we can link to an openapi visualisation of the schema, but even the raw file might be ok).

As far as the example values go, why do you say they are not valid, can you point to something specific? I generally feel those could be useful for users to copy and then substitute the values they need in

Reply:
> I think inlining the schema (at least fully-resolved schemas) is a no-go

OK. Not resolving is probably fine.